cartoframes/viz/layer.py

```diff
@@ -149,3 +160,34 @@ def _set_legend(legend):
         return legend
     else:
         return Legend()
+
+
+def _set_widgets(widgets):
+    """Set a Widget class from the input"""
+    if isinstance(widgets, list):
+        return list(map(_set_widgets, widgets))
```

Comment:
We can create a `Widgets` class to manage the list case, so we can clean the code a bit (`this.widgets.get_variables()`, etc) and move `_get_widgets_variables` and `_get_widget_info` from the Layer class